contracts/StrategyCurve3crv.sol

```diff
@@ -288,27 +334,56 @@ contract StrategyCurve3crv is StrategyCurveBase {
         }
 
         // we're done harvesting, so reset our trigger if we used it
-        forceHarvestTriggerOnce = false;
+        delete forceHarvestTriggerOnce;
+    }
+
+    /* ========== ySwaps ========== */
+
+    /** 
+    @notice Set ySwaps address, approve ySwaps contract to sell our reward tokens 
+    and register the swaps required by this strategy.
+    @param _tradeFactory ySwaps contract address
+    */
+    function setTradeFactory(address _tradeFactory) external onlyGovernance {
+        require(_tradeFactory != address(0), "!address");
+
+        _removeTradeFactoryPermissions();
+
+        // approve and set up trade factory
+        crv.safeApprove(_tradeFactory, type(uint256).max);
+        gno.safeApprove(_tradeFactory, type(uint256).max);
+
+        ITradeFactory(_tradeFactory).enable(address(crv), address(want));
```

Comment:
As the PR stands, lines 356 and 357 need to be commented out for all the tests to pass. This is only temporary as I'll update the PR once we have the trade handler contract deployed. That way, I can re-run the tests to validate that they still pass, which they should.